File: frontend/src/lib/types/media.ts
export interface Media {
  ID: number;
  FileName: string;
  OriginalName: string;
  FilePath: string;
  ThumbnailPath: string;
  FileSize: number;
  MimeType: string;
  Width?: number;
  Height?: number;
  AltText: string;
  EntityType: string;
  EntityID?: number;
  CreatedAt?: string;
  UpdatedAt?: string;
}

export interface MediaUploadResponse {
  ID: number;
  FileName: string;
  OriginalName: string;
  FilePath: string;
  ThumbnailPath: string;
  FileSize: number;
  MimeType: string;
  Width?: number;
  Height?: number;
  AltText: string;
  EntityType: string;
  EntityID?: number;
  CreatedAt: string;
  UpdatedAt: string;
}

File: backend/internal/domain/media.go
package domain

import "time"

type MediaType string

const (
	MediaTypeImage    MediaType = "image"
	MediaTypeVideo    MediaType = "video"
	MediaTypeDocument MediaType = "document"
)

type Media struct {
	ID            uint
	FileName      string
	OriginalName  string
	FilePath      string
	ThumbnailPath string
	FileSize      int64
	MimeType      string
	Width         *int
	Height        *int
	AltText       string
	EntityType    string // "product", "category", etc.
	EntityID      *uint
	CompanyID     uint // ID da empresa/tenant (obrigatório - Sprint 3)
	DeletedAt     *time.Time
	CreatedAt     time.Time
	UpdatedAt     time.Time
}
